qt-cao-editor/app/scenemodifier.cpp

```diff
@@ -0,0 +1,521 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 Vikas Thamizharasan
+****************************************************************************/
+
+#include "scenemodifier.h"
+
+#include <QtCore/QDebug>
+
+#include <QMaterial>
+#include <QEffect>
+
+#include <QTechnique>
+#include <QRenderPass>
+#include <QShaderProgram>
+
+
+SceneModifier::SceneModifier(Qt3DCore::QEntity *rootEntity, QWidget *parentWidget)
+    : m_rootEntity(rootEntity)
+    , m_parentWidget(parentWidget)
+    , m_lineEntity(nullptr)
+    , m_cylinderEntity(nullptr)
+    , m_circleEntity(nullptr)
+{
+    // World Axis
+    this->createLines(QVector3D(-100000.0f, 0.0f, 0.0f), QVector3D(100000.0f, 0.0f, 0.0f), 0, true, ""); //X-axis
+    this->createLines(QVector3D(0.0f, -100000.0f, 0.0f), QVector3D(0.0f, 100000.0f, 0.0f), 1, true, ""); //Y-axis
+    this->createLines(QVector3D(0.0f, 0.0f, -100000.0f), QVector3D(0.0f, 0.0f, 100000.0f), 2, true, ""); //Z-axis
```

Comment:
Commenting these 3 lines solve somehow the issue with `QObjectPicker` on my computer.